templates/todo/projects/nodejs-mongo/.repo/terraform/infra/main.tfvars.json

```diff
@@ -1,5 +1,6 @@
 {
     "location": "${AZURE_LOCATION}",
     "environment_name": "${AZURE_ENV_NAME}",
-    "principal_id": "${AZURE_PRINCIPAL_ID}"
+    "principal_id": "${AZURE_PRINCIPAL_ID}",
+    "useAPIM" : "${USE_APIM}"

```

Comment:
This is not going to work unless the user sets the `USE_APIM` environment variable, otherwise terraform will fail on an empty string when a boolean variable is required.

We should consider removing this variable in the tfvars.json by default and have instructions to set it when they want to use APIM.

Reply:
@wbreza, @weikanglim  

We currently have some ideas about setting the value of the `USE_APIM`:

1.Set the `USE_APIM` to type `string`.
-  With this approach, CI will not have any errors, and user can set `USE_APIM` when they need it.

2.Set the `USE_APIM` to type `bool`. 
- The first case: removing `USE_APIM` in the `tfvars.json` by default. If the user needs to use `apim`, he must manually set this value to `true/false` in the `variables.tf` file.<img width="686" alt="image" src="https://user-images.githubusercontent.com/81678720/210966507-caa0ff04-e5ce-4faa-9476-de495e160ac4.png"> 
- The second case: if you do not remove the parameter `USE_APIM` in the `tfvars.json`, the user must set the value of the `USE_APIM` before running, otherwise an error will be reported. And CI will fail.
- The third case: this method involve the underlying design of `azd`, let the user set the value `USE_APIM` when executing `azd init`, and set a default value for it. Would you consider doing this?

Which way do you prefer?
Besides, currently our pr has set `USE_APIM` to `string` type, and both CI and tests run fine.

Reply:
@zedy-wj I'm working on these changes right now. For terraform, we can set the variable as `bool`, and configure `tfvars.json` to contain:  `${USE_APIM=false}` which provides a default value of `false` if the env variable is unset.

Reply:
@weikanglim - We have updated according to your comments, it works fine, thanks for your suggestion. Besides, this method does not apply to templates other than Terraform, and we still use the `string` type for the `USE_APIM` parameters of other templates, please re-review it, thanks.